tests/integ/sagemaker/jumpstart/conftest.py

```diff
@@ -113,6 +133,35 @@ def _teardown():
     bucket = s3_resource.Bucket(test_cache_bucket)
     bucket.objects.filter(Prefix=test_suite_id + "/").delete()
 
+    # delete private hubs
+    _delete_hubs(sagemaker_session)
+
+
+def _delete_hubs(sagemaker_session):
+    # list Hubs created by PySDK integration tests
+    list_hub_response = sagemaker_session.list_hubs(name_contains=HUB_NAME_PREFIX)
+
+    for hub in list_hub_response["HubSummaries"]:
+        if hub["HubName"] != SM_JUMPSTART_PUBLIC_HUB_NAME:
```

Comment:
While we don't delete public hub, should we also restrict to only delete the hubs that are associated with a pysdk integ test? This can delete the hubs we don't want to in the account that runs the integ test

Reply:
I am already filtering the hubs related only to PySDK integ test on line 142
`sagemaker_session.list_hubs(name_contains=HUB_NAME_PREFIX)`

Reply:
Concurrent runs of the test will cause an issue with this clean-up strategy. Should it clean-up the hub created by this specific test?

Reply:
yeah i actually had to change it later on, I have updated the commit

Reply:
isn't the suite id unique to each run? That would prevent overlapping test sessions on the same account/region from interfering with each other.

Reply:
@evakravi yes but I wasn't deleting the unique Hub created by a specific Pytest session. I was rather deleting all the hubs created by PySDK test regardless of whether it was created by a specific pytest session or not. PySDK runs this command to run integ tests `pytest tests/integ -m 'not local_mode and not cron and not slow_test' -n 240` which means there are 240 parallel workers and each would corresponds to a separate pytest session. Because I am not cleaning up specific test session created hub, this will delete the ones which are being used by other pytest sessions and tests in parallel.